tests/test_sqs/test_sqs.py

```diff
@@ -3115,3 +3115,44 @@ def test_message_has_windows_return():
     messages = queue.receive_messages()
     messages.should.have.length_of(1)
     messages[0].body.should.match(message)
+
+
+@mock_sqs
+def test_message_delay_is_more_than_15_minutes():
+    client = boto3.client("sqs", region_name="us-east-1")
+    response = client.create_queue(
+        QueueName=f"{str(uuid4())[0:6]}.fifo", Attributes={"FifoQueue": "true"}
+    )
+    queue_url = response["QueueUrl"]
+
+    msgs = client.send_message_batch(
+        QueueUrl=queue_url,
+        Entries=[
+            {
+                "Id": "1",
+                "MessageBody": "foo",
+                "DelaySeconds": 180,
+                "MessageGroupId": "message_group_id_1",
+            },
+            {
+                "Id": "2",
+                "MessageBody": "foo",
+                "DelaySeconds": 1800,
+                "MessageGroupId": "message_group_id_1",
+            },
+            # {
+            #     "Id": "3",
+            #     "MessageBody": "foo",
+            #     "DelaySeconds": 100,
+            #     "MessageGroupId": "message_group_id_1",
+            # },
+            # {
+            #     "Id": "4",
+            #     "MessageBody": "foo",
+            #     "DelaySeconds": 2800,
+            #     "MessageGroupId": "message_group_id_1",
+            # },
+        ],
+    )
+
+    print(msgs)
```

Comment:
Can we add some assertions here, that `msgs` contains the correct response?

Reply:
for this part, what exact assertions you are expecting. Can you please elaborate

Reply:
I have done something like this:

@mock_sqs
def test_message_delay_is_more_than_15_minutes():
    client = boto3.client("sqs", region_name="us-east-1")
    response = client.create_queue(
        QueueName=f"{str(uuid4())[0:6]}.fifo", Attributes={"FifoQueue": "true"}
    )
    queue_url = response["QueueUrl"]

    response = client.send_message_batch(
        QueueUrl=queue_url,
        Entries=[
            {
                "Id": "id_1",
                "MessageBody": "body_1",
                "DelaySeconds": 10,
                "MessageAttributes": {
                    "attribute_name_1": {
                        "StringValue": "attribute_value_1",
                        "DataType": "String",
                    }
                },
                "MessageGroupId": "message_group_id_1",
                "MessageDeduplicationId": "message_deduplication_id_1",
            },
            {
                "Id": "id_2",
                "MessageBody": "body_2",
                "DelaySeconds": 1800,
                "MessageAttributes": {
                    "attribute_name_2": {"StringValue": "123", "DataType": "Number"}
                },
                "MessageGroupId": "message_group_id_2",
                "MessageDeduplicationId": "message_deduplication_id_2",
            },
        ],
    )

    sorted([entry["Id"] for entry in response["Successful"]]).should.equal(["id_1"])

    sorted([entry["Id"] for entry in response["Failed"]]).should.equal(["id_2"])

    print(response)

    time.sleep(11)

    response = client.receive_message(
        QueueUrl=queue_url,
        MaxNumberOfMessages=10,
        MessageAttributeNames=["attribute_name_1", "attribute_name_2"],
        AttributeNames=["MessageDeduplicationId", "MessageGroupId"],
    )

    response["Messages"].should.have.length_of(1)
    response["Messages"][0]["Body"].should.equal("body_1")
    response["Messages"][0]["MessageAttributes"].should.equal(
        {"attribute_name_1": {"StringValue": "attribute_value_1", "DataType": "String"}}
    )
    response["Messages"][0]["Attributes"]["MessageGroupId"].should.equal(
        "message_group_id_1"
    )
    response["Messages"][0]["Attributes"]["MessageDeduplicationId"].should.equal(
        "message_deduplication_id_1"
    )

I am receiving the message using receive_message function and then checking its attributes. Here the problem is - if delay seconds is i then we have to wait for i+1 seconds to receive the message then only we can do some checking. Is it ok or you are expecting something else

Reply:
Yes, I think that's fine. 

To speed things up, we can just reduce the DelaySeconds-parameter - e.g., if we can set it to 2, so we only have to wait 3 seconds.

Reply:
OK sir I have raised PR with required changes. Let me know if it works